### PR TITLE
Create a test case for format table

### LINKED ISSUE #25: Create tests/test_report_generator.py test_format_table()

### PR DESCRIPTION
**Pull request by ameliav**
#  #

## 🗣 Description ##

This PR resolves issue https://github.com/cisagov/tpt-reports/issues/25 and creates a test case for the format_table function in report_generator.py.

## 💭 Motivation and context ##

This will add a test case to an untested function and give more coverage.

## 🧪 Testing ##

All automated tests, including this new test, were successful.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

**Review by schmelz21 — APPROVED** [review]

File: tests/test_report_generator.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    column_widths = [inch, inch, inch, inch, inch, inch, inch, inch]
+    column_style_list = [body, body, body, body, body, body, body, body]
+
+    # Call the format_table function with test data
+    table = report_generator.format_table(
+        test_dataframe, header_style, column_widths, column_style_list
+    )
+    # Perform assertions to check the table's properties
+    assert isinstance(test_dataframe, pd.DataFrame)
+    assert isinstance(table, Table)
```
Comment (dav3r) [inline tests/test_report_generator.py:40]: Can any attributes of the generated table be validated?  That would make for a more thorough test.

**Comment by dav3r** [comment]
Also, the "Testing" section of this PR's description is blank.  You should always include some explanation of how the changes were tested, even it's something simple like "All automated tests, including this new test, were successful."

File: tests/test_report_generator.py (hunk 1 above)
Comment (ameliav) [reply tests/test_report_generator.py:40]: Additional assertions validating attributes have been added here 4a9b13aea6f91ac0a213c946c1578f0c2dc85ca5 along with assertions that test row & col length. Please let me know if this is what you had in mind, thanks.

Comment (dav3r) [reply tests/test_report_generator.py:40]: Yes, that's exactly the type of thing I was looking for. 👍

File: tests/test_report_generator.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    table = report_generator.format_table(
+        test_dataframe, header_style, column_widths, column_style_list
+    )
+
+    # Perform assertions to check the table's properties
+    assert isinstance(test_dataframe, pd.DataFrame)
+    assert isinstance(table, Table)
+
+    # Validate the number of rows and columns in the table
+    assert table._nrows == len(test_dataframe) + 1
```
Comment (dav3r) [inline tests/test_report_generator.py:44]: Can you please add a brief comment explaining why the "+ 1" is needed here?

Comment (ameliav) [reply tests/test_report_generator.py:44]: Resolved in 9f357f6473f410267f3391f56fe678af8a0504fd

Comment (dav3r) [reply tests/test_report_generator.py:44]: Thanks!

**Review by dav3r — APPROVED** [review]
Nice work! 👍

**Pull request merged**